app/uk/gov/hmrc/gform/core/parsers/ValueParser.scala

```diff
@@ -329,8 +329,8 @@ trait ValueParser extends RegexParsers with PackratParsers with BasicParsers {
     | "concat(" ~ _expr1 ~ (("," ~> _expr1) *) ~ ")" ^^ { case _ ~ expr ~ exprs ~ _ =>
       Concat(expr +: exprs)
     }
-    | "choice(" ~ FormComponentId.unanchoredIdValidation ~ ")" ^^ { case _ ~ value ~ _ =>
-      ChoiceLabel(FormComponentId(value))
+    | "choice(" ~ _expr1 ~ (("," ~> _expr1) *) ~ ")" ^^ { case _ ~ expr ~ exprs ~ _ =>
+      ChoiceLabel(expr +: exprs)
```

Comment:
```suggestion
    | "choice(" ~ FormComponentId.unanchoredIdValidation ~ ")" ^^ { case _ ~ value ~ _ =>
      Concat(FormCtx(FormComponentId(value)) :: Nil)
```

We should not change the parser.

We should remove `final case class ChoiceLabel(formComponentId: FormComponentId) extends Expr` altogether.